src/ontoweaver/make_labels.py

```diff
@@ -0,0 +1,176 @@
+import logging
+import abc
+import re
+
+from . import errormanager, exceptions
+
+logger = logging.getLogger("ontoweaver")
+
+class ReturnCreate:
+    """Class for returning the extracted cell value of the transformer, along with the edge type, target node type,
+    and properties of the target node type.
+    """
+
+    def __init__(self, extracted_cell_value = None, edge_type = None, target_node_type = None, properties_of = None):
+        """
+        Initializes the ReturnCreate object.
+
+        Args:
+            extracted_cell_value: The extracted cell value of the transformer. Can serve as ID of the node, ID of the edge, or value of property.
+            edge_type: The type of the edge.
+            target_node_type: The type of the target node.
+            properties_of: The properties of the returned target node type.
+        """
+
+        self.extracted_cell_value = extracted_cell_value # Can serve as ID of the node, ID of the edge, or value of property.
+        self.edge_type = edge_type
+        self.target_node_type = target_node_type
+        self.target_element_properties = properties_of
+
+class LabelMaker(errormanager.ErrorManager, metaclass=abc.ABCMeta):
+    """Interface for defining the correct labels (types) for each extracted value.
+    The class is responsible for validating the extracted cell value and returning the appropriate label.
+    """
+
+    def __init__(self, raise_errors: bool = True):
+        """
+        Initializes the LabelMaker object.
+
+        Args:
+            raise_errors (bool): if True, will raise an exception when an error is encountered, else, will log the error and try to proceed.
+        """
+        super().__init__(raise_errors)
+
+    def check_attributes(self, **kwargs):
```

Comment:
This function may not be necessary if `kwargs` goes through `__init__` and Python handle member variables initialization checks.